app/main/authorisation.py

```diff
@@ -0,0 +1,31 @@
+import pandas as pd
+
+from app.const import ORGANISATION_ID_TO_PLACE_NAMES
+from config import Config
+
+
+def get_local_authority_place_names(user_email: str) -> None | list[str]:
+    """
+    Get the local authority place names corresponding to a user's email domain.
+
+    This function takes a user's email address and uses the domain part (after '@')
+    to look up the corresponding local authority from a CSV file. If a matching local
+    authority is found, its identifier is returned, and this is used to look up the
+    place names for which the local authority is expected to submit for.
+
+    :param user_email: A string representing the user's email address.
+    :return: An array of place names or None if none are found.
+    """
+    lookup_url = Config.LOCAL_AUTHORITIES_LOOKUP
+    df_lookup = pd.read_csv(lookup_url, index_col=0)
+
+    organisation_id = df_lookup[df_lookup["homepage_url"].str[11:-1] == user_email.split("@")[1]]["gss"]
```

Comment:
is .str[11:-1] intended to remove the https protocal from the front the url. If so to deal with http(s)

i.e:
https://www.monmouthshire.gov.uk/
http://www.moray.gov.uk

could we split on . instead?

Reply:
Yes, we might be able to. I'll have a look at the list and see if there's any exceptions, and if not re-implement by splitting on the "." char and selecting the right portion of the array if we decide to continue with this approach of authorising via e-mail domain.

Reply:
Thinking about it, we can't split on the "." char because some urls have a "/" at the end and some don't, which is how I arrived at the logic for 11:-1 vs. 11:, but we can just generically remove the https protocol and any trailing slashes by default and then do the logic on the dataframe lookup. That will save us from having to do the lookup logic twice over.

Reply:
Could we use an external library, as per other comment, or use `netloc` from `urlparse` in the core library, which should give us a more standardised string?